src/jira/client/axios.ts

```diff
@@ -18,13 +18,30 @@ export class JiraClientError extends Error {
 	status?: number;
 	cause: AxiosError;
 
-	constructor(message: string, cause: AxiosError, status?: number) {
+	constructor(
+		message: string,
+		cause: AxiosError,
+		status: number | undefined
```

Comment:
why is `status: number | undefined` preffered over `status?: number`

Reply:
So this can make sure `new JiraError("blah", err)` fail the TS check while `new JiraError("blah", err, someStatusVariable)` pass. 

So that when constructing a JiraRateLimitError, you have to remember to pass in a status, even it can be undefined.